packages/flutter/lib/src/rendering/flex.dart

```diff
@@ -12,6 +12,84 @@ import 'layer.dart';
 import 'layout_helper.dart';
 import 'object.dart';
 
+// A type that describes a 2D size along the main axis and the cross axis of a
```

Comment:
Overall, I found the sizing and positioning algorithm way harder to follow with the _AxisSize and _LayoutAxisDimensions abstractions in place - compared to what it was before. Part of the problem is that after reading everything I couldn't explain what an _AxisSize or a _LayoutAxisDimensions _is_, which makes me think that these are not helpful abstractions?